L1Trigger/TrackFindingTracklet/test/AnalyzerTracklet.cc

```diff
@@ -0,0 +1,247 @@
+    // DataFormats token
+    ESGetToken<DataFormats, DataFormatsRcd> esGetTokenDataFormats_;
+    // stores, calculates and provides run-time constants
+    const Setup* setup_;
+    // helper class to extract structured data from TTDTC::Frames
+    const DataFormats* dataFormats_;
+    // enables analyze of TPs
+    bool useMCTruth_;
+    //
+    int nEvents_;
+
+    // Histograms
+
+    TProfile* prof_;
+    TProfile* profChannel_;
+    TH1F* hisChannel_;
+    TH1F* hisEff_;
+    TH1F* hisEffTotal_;
+    TEfficiency* eff_;
+
+    // printout
+    stringstream log_;
+  };
+
+  AnalyzerTracklet::AnalyzerTracklet(const ParameterSet& iConfig) : useMCTruth_(iConfig.getParameter<bool>("UseMCTruth")), nEvents_(0) {
+    usesResource("TFileService");
+    // book in- and output ED products
+    const InputTag& inputTag = iConfig.getParameter<InputTag>("InputTag");
+    edGetToken_ = consumes<TTTracks>(inputTag);
+    if (useMCTruth_) {
+      const auto& inputTagSelecttion = iConfig.getParameter<InputTag>("InputTagSelection");
+      const auto& inputTagReconstructable = iConfig.getParameter<InputTag>("InputTagReconstructable");
+      edGetTokenSelection_ = consumes<StubAssociation>(inputTagSelecttion);
+      edGetTokenReconstructable_ = consumes<StubAssociation>(inputTagReconstructable);
+    }
+    // book ES products
+    esGetTokenSetup_ = esConsumes<Setup, SetupRcd, Transition::BeginRun>();
+    esGetTokenDataFormats_ = esConsumes<DataFormats, DataFormatsRcd, Transition::BeginRun>();
+    // initial ES products
+    setup_ = nullptr;
+    dataFormats_ = nullptr;
+    // log config
+    log_.setf(ios::fixed, ios::floatfield);
+    log_.precision(4);
+  }
+
+  void AnalyzerTracklet::beginRun(const Run& iEvent, const EventSetup& iSetup) {
+    // helper class to store configurations
+    setup_ = &iSetup.getData(esGetTokenSetup_);
+    // helper class to extract structured data from TTDTC::Frames
+    dataFormats_ = &iSetup.getData(esGetTokenDataFormats_);
+    // book histograms
+    Service<TFileService> fs;
+    TFileDirectory dir;
+    dir = fs->mkdir("Tracklet");
+    prof_ = dir.make<TProfile>("Counts", ";", 9, 0.5, 9.5);
+    prof_->GetXaxis()->SetBinLabel(1, "Stubs");
+    prof_->GetXaxis()->SetBinLabel(2, "Tracks");
+    prof_->GetXaxis()->SetBinLabel(3, "Lost Tracks");
+    prof_->GetXaxis()->SetBinLabel(4, "Matched Tracks");
+    prof_->GetXaxis()->SetBinLabel(5, "All Tracks");
+    prof_->GetXaxis()->SetBinLabel(6, "Found TPs");
+    prof_->GetXaxis()->SetBinLabel(7, "Found selected TPs");
+    prof_->GetXaxis()->SetBinLabel(8, "Lost TPs");
+    prof_->GetXaxis()->SetBinLabel(9, "All TPs");
+    // channel occupancy
+    constexpr int maxOcc = 180;
+    const int numChannels = setup_->numRegions();
+    hisChannel_ = dir.make<TH1F>("His Channel Occupancy", ";", maxOcc, -.5, maxOcc - .5);
+    profChannel_ = dir.make<TProfile>("Prof Channel Occupancy", ";", numChannels, -.5, numChannels - .5);
```

Comment:
"Prof channel occupancy" should be "No. of tracks per nonant". (There is more than one channel per nonant, so "channel" seems wrong. And without "tracks", the title may refer to stubs).